File: 백준/Bronze/31994. 강당 대관/강당 대관.py
import sys
input=lambda:sys.stdin.readline().rstrip()
seminar=[]
for _ in range(7):
    sbj,num=input().split()
    seminar+=[(sbj,int(num))]
seminar=sorted(seminar,key=lambda x:(x[1],x[0]),reverse=True)
print(seminar[0][0])

File: 백준/Bronze/31994. 강당 대관/강당 대관.cc
#include <iostream>
#include <vector>
using namespace std;
int main(void) {
    string sbj, r;
    int num, maxn = 0;
    for (int i = 0; i < 7; i++) {
        cin >> sbj >> num;
        if (maxn < num) {
            maxn = num;
            r = sbj;
        }
    }
    cout << r;
    return 0;
}